core/blockchain.go

```diff
@@ -559,6 +564,49 @@ func (bc *BlockChain) initHvmHeaderNode(config *tbc.Config) {
 	bc.hvmEnabled = true
 }
 
+func (bc *BlockChain) SetupDeucalion(pctx context.Context, address string) error {
+	d, err := deucalion.New(&deucalion.Config{
+		ListenAddress: address,
+	})
+	if err != nil {
+		return err
+	}
+
+	ctx, cancel := context.WithCancel(pctx)
+
+	go func() {
+		defer cancel()
+		if err := d.Run(ctx, nil, func(ctx context.Context) (bool, any, error) {
+			return bc.healthyNode.Load(), bc.CurrentBlock(), nil
+		}); !errors.Is(err, context.Canceled) {
```

Comment:
I like nesting err assignments most of the time, but this is a little hard to read:
```suggestion
		err := d.Run(ctx, nil, func(ctx context.Context) (bool, any, error) {
			return bc.healthyNode.Load(), bc.CurrentBlock(), nil
		})
		if err != nil && !errors.Is(err, context.Canceled) {
```

Reply:
Agree 👍